### ISSUE TITLE
dbus_variant_var macro is visibility issue.

### ISSUE THREAD

**Issue by cmaves** [KillingSpark/rustbus#38]
The dbus_variant_var is broken when you try to use in an external crate. You get this error:
```
error[E0451]: field `byteorder` of struct `rustbus::wire::unmarshal::traits::Variant` is private
  --> src/gatt/mod.rs:26:1
   |
26 | dbus_variant_var!(CharVar, U16 => u16; String => String);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ private field
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
```
I believe this happens because the macro's generated code is has the same scope as the external crate that is its being generated in. So if the macro's generated code references a private member or function then it will fail to compile.

The two solutions I could think of would be to: 
1. The simplest Use `Variant`'s unmarshal function rather build the Variant. This comes with a performance penalty as it parses the signature twice.
2. Create and expose a function for `Variant` to construct the Variant from the signature, byteorder, and buffer.

It might also be worth it to move the macro tests into an integration test so that visibility issues can be caught ahead of time. Thoughts?

**Comment by KillingSpark** [comment]
Seems like I forgot to add this to the examples, that would have catched this error.  In the refactoring I am doing currently I already fixed that 'by accident' by using your first suggestions and just unmarshal it. Parsing the signature twice is probably not much in the big picture.

I'll fix this and make a new release.

**Comment by KillingSpark** [comment]
Should be fixed with 75b51ee. (release v0.7.1)

**Comment by cmaves** [comment]
Everything seems to be working now!
